edb/schema/indexes.py

```diff
@@ -41,7 +42,11 @@
     from . import types as s_types
 
 
-class Index(referencing.ReferencedInheritingObject, s_anno.AnnotationSubject):
+class Index(
+    referencing.ReferencedInheritingObject,
+    s_anno.AnnotationSubject,
+    qlkind=qltypes.SchemaObjectClass.INDEX,
```

Comment:
Can you add some check to the metaclass to require `qlkind` for "final" classes? Perhaps we can add something like `__abstract__ = True` for non-final ones.